src/lib/terkin/lib/hx711.py

```diff
@@ -121,11 +121,11 @@ def read(self):
         # Shift in data, gain & channel info.
         result = 0
         for j in range(24 + self.GAIN):
-            state = disable_irq()
+            #state = disable_irq()
             self.pSCK(True)
             self.pSCK(False)
+            #enable_irq(state)
             result = (result << 1) | self.pOUT()
-            enable_irq(state)
```

Comment:
Is this sane? Does disabling the interrupts interfere with the dualchannel-reading somehow?

Reply:
I ran tests and disabling the irq did nothing to speed it up. It would be faster if the irqs are disabled for the whole for loop but then the MCU crashes. 
So its useless and potentially adds instability.

Reply:
> I ran tests and disabling the irq did nothing to speed it up. It would be faster if the irqs are disabled for the whole for loop but then the MCU crashes.

Disabling interrupts is not about gaining more speed. It's about safety.

> So its useless and potentially adds instability.

I can't follow here and have strong objections against not disabling interrupts while reading the pin. It's also in the original driver:

https://github.com/geda/hx711-lopy/blob/9cc6de8d878abfe3be6a3fb43e4b341f816a4952/hx711.py#L41-L46

Reply:
> In the description of the C++ driver it is also explicitly mentioned how important it is to protect the writing of the clock signal from interrupts. So I withdraw my remark that `disableirq()` is useless (I also take it out of the PR).
>
> -- https://community.hiveeyes.org/t/gewichtspeaks-im-lte-betrieb/3279/10

Thanks!